File: src/js/src/scopes/MultiLineDefinitionScope.js
function MultiLineDefinitionScope() {
	this.type = ScopeTypeCache.multiLineDefinition;
	this.level = ScopeLevel.block;
	this.canNestText = true;
	this.canNestParagraph = true;
	
	this.open = function(analyzer) {
		analyzer.openMultiLineDefinition();
	};
	this.close = function(analyzer) {
		analyzer.closeMultiLineDefinition();
	};
}

extend(Scope, MultiLineDefinitionScope);

File: src/dotnet/Src/ButterflyNet.Parser/Scopes/MultiLineDefinitionScope.cs
﻿namespace ButterflyNet.Parser.Scopes {
	public class MultiLineDefinitionScope : BlockScope {
		public override bool CanNestParagraph { get { return true; } }

		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenMultiLineDefinition();
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseMultiLineDefinition();
		}
	}
}